daml-foundations/daml-tools/daml-cli/DA/Cli/Visual.hs

```diff
@@ -54,15 +55,18 @@ startFromExpr seen world e = case e of
 startFromChoice :: LF.World -> LF.TemplateChoice -> Set.Set Action
 startFromChoice world chc = startFromExpr Set.empty world (LF.chcUpdate chc)
 
-templatePossibleUpdates :: LF.World -> LF.Template -> Set.Set Action
-templatePossibleUpdates world tpl = Set.unions $ map (startFromChoice world) (NM.toList (LF.tplChoices tpl))
+data ChoiceAndAction = ChoiceAndAction { choiceForTemplate :: LF.Template , choice :: LF.TemplateChoice ,actions :: Set.Set Action }
+data TemplateChoiceAction = TemplateChoiceAction { template :: LF.Template ,choiceAndAction :: [ChoiceAndAction] }
 
-moduleAndTemplates :: LF.World -> LF.Module -> [(LF.TypeConName, Set.Set Action)]
+
+templatePossibleUpdates :: LF.World -> LF.Template -> [ChoiceAndAction]
+templatePossibleUpdates world tpl = map (\c -> ChoiceAndAction tpl c (startFromChoice world c)) (NM.toList (LF.tplChoices tpl))
+
+moduleAndTemplates :: LF.World -> LF.Module -> [TemplateChoiceAction]
 moduleAndTemplates world mod = retTypess
     where
         templates = NM.toList $ LF.moduleTemplates mod
-        retTypess = map (\t-> (LF.tplTypeCon t, templatePossibleUpdates world t )) templates
-
+        retTypess = map (\t-> TemplateChoiceAction t (templatePossibleUpdates world t ) ) templates
```

Comment:
```suggestion
        retTypess = map (\t -> TemplateChoiceAction t (templatePossibleUpdates world t)) templates
```
You're spaces have a tendency to appear in strange places. I suppose this is a result of years of scala format?

Reply:
bad habits from using intellij auto format thing :(